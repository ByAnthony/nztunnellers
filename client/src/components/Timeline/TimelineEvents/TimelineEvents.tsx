import { TimelineEvent } from '../TimelineEvent/TimelineEvent';
import {
  Event, MilitaryYears, PostServiceYears,
} from '../../../types/tunneller';

import STYLES from '../Timeline.module.scss';

type Props = {
    militaryYears: MilitaryYears,
    postServiceYears: PostServiceYears,
}

export function TimelineEvents({ militaryYears, postServiceYears }: Props) {
  const { frontEvents } = militaryYears;

  const place = () => {
    const location = militaryYears.endOfService.deathWar?.place?.location;
    const town = militaryYears.endOfService.deathWar?.place?.town;
    if (location && town) {
      return `${location}, ${town}`;
    }
    if (location && !town) {
      return location;
    }
    return null;
  };

  const disease = militaryYears.endOfService.deathWar?.cause?.circumstances;

  const warInjuries = postServiceYears.death?.cause?.circumstances;

  return (
    <>
      {frontEvents.map((event: Event) => (
        <div key={frontEvents.indexOf(event)}>
          <div className={STYLES.date}>
            <div className={STYLES.year}>{`${event.date.year}`}</div>
            <div className={STYLES['day-month']}>{`${event.date.dayMonth}`}</div>
          </div>
          <div className={STYLES['event-wrapper']}>
            <TimelineEvent
              event={event.event}
              place={place}
              disease={disease}
              warInjuries={warInjuries}
            />
          </div>
        </div>
      ))}
    </>
  );
}
