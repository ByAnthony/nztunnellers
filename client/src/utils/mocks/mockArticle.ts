import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
