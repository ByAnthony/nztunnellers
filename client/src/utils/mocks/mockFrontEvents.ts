import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
