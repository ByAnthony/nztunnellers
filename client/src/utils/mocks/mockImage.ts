import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'Courtesy of John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook: ImageBook = {
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: 'p. 89',
};

export const mockImageSource: ImageSource = {
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
};

export const mockImage: Image = {
  url: '1/1234.jpg',
  source: mockImageSource,
};
