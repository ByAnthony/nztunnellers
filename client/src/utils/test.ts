import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ReactElement } from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';

export const findElementWithText = (text: string) => {
  const elements = screen.getAllByText((content) => content.includes(text));

  if (elements.length === 0) {
    throw new Error(`Unable to find an element with the text: ${text}`);
  }

  return elements[0];
};

export const renderWithMemoryRouter = (ui: ReactElement, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);
  const user = userEvent.setup();

  return {
    user,
    ...render(ui, { wrapper: MemoryRouter }),
  };
};

const renderWithRouter = (ui: ReactElement, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);
  const user = userEvent.setup();

  return {
    user,
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

export * from '@testing-library/react';
export { renderWithRouter as render };
