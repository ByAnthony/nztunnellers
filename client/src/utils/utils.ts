export const displayBirthDeathDates = (
  birth: string | null,
  death: string | null,
) => {
  if (birth !== null && death !== null) {
    return `${birth} - ${death}`;
  }
  if (birth !== null && death === null) {
    return `${birth} - †?`;
  }
  if (birth === null && death !== null) {
    return `? - ${death}`;
  }
  return '? - †?';
};
