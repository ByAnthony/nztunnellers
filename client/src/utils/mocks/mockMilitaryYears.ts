import {
  Date,
  DeathPlace,
  Death,
  DeathCause,
  Cemetery,
  Medal,
  MilitaryYears,
  Enlistment,
  EmbarkationUnit,
  Training,
  Transport,
  EndOfService,
  Demobilization,
} from '../../types/tunneller';
import { mockFrontEvents } from './mockFrontEvents';

export const mockBritishWarMedal: Medal = {
  name: 'British War Medal',
  country: 'United Kingdom',
  image: 'british-war-medal.png',
  citation: null,
};

export const mockVictoryMedal: Medal = {
  name: 'Victory Medal',
  country: 'United Kingdom',
  image: 'victory-medal.png',
  citation: null,
};

export const mockMedals: Medal[] = [
  mockBritishWarMedal,
  mockVictoryMedal,
];

export const mockDate: Date = {
  year: '1926',
  dayMonth: '13 October',
};

const mockPlace: DeathPlace = {
  location: 'Telegraph Hill',
  town: 'Arras',
  country: 'France',
};

const mockCause: DeathCause = {
  type: 'Killed in action',
  circumstances: 'Dead wounded by shell fire',
};

const mockCemetery: Cemetery = {
  name: 'Faubourg d\'Amiens',
  location: 'Arras',
  country: 'France',
  graveReference: 'IE 18',
};

export const mockDeath: Death = {
  date: mockDate,
  place: mockPlace,
  cause: mockCause,
  cemetery: mockCemetery,
  ageAtDeath: 89,
};

export const mockEnlistment: Enlistment = {
  serial: '1/1000',
  rank: 'Sapper',
  date: mockDate,
  district: 'Canterbury',
  alias: null,
  transferredToTunnellers: null,
};

const mockTraining: Training = {
  date: mockDate,
  location: 'Auckland',
  locationType: 'Military Camp',
};

export const mockEmbarkationUnit: EmbarkationUnit = {
  detachment: 'Main Body',
  training: mockTraining,
  section: 'Section No.2',
  attachedCorps: null,
};

export const mockTransport: Transport = {
  reference: 'HMNZT 56',
  vessel: 'Beata',
  departureDate: mockDate,
  departurePort: 'Auckland',
  arrivalDate: mockDate,
  arrivalPort: 'Plymouth',
};

const mockDemobilization: Demobilization = {
  date: mockDate,
  country: 'New Zealand',
};

const mockEndOfService: EndOfService = {
  deserter: false,
  transferred: null,
  deathWar: null,
  transportNz: mockTransport,
  demobilization: mockDemobilization,
};

export const mockMilitaryYears: MilitaryYears = {
  enlistment: mockEnlistment,
  embarkationUnit: mockEmbarkationUnit,
  transportUk: mockTransport,
  frontEvents: mockFrontEvents,
  endOfService: mockEndOfService,
  medals: mockMedals,
};
