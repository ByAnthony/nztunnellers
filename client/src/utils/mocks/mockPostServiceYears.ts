import { PostServiceYears, PostWarDeath } from '../../types/tunneller';
import { mockDeath } from './mockMilitaryYears';

export const mockAfterWarDeath: PostWarDeath = {
  deathWarInjury: false,
  ...mockDeath,
};

export const mockPostServiceYears: PostServiceYears = {
  death: mockAfterWarDeath,
};
