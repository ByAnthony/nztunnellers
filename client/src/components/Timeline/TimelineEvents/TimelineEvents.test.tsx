import '@testing-library/jest-dom/extend-expect';

import { mockMilitaryYears } from '../../../utils/mocks/mockMilitaryYears';
import { mockPostServiceYears } from '../../../utils/mocks/mockPostServiceYears';
import { render } from '../../../utils/test';

import { TimelineEvents } from './TimelineEvents';

test('should render TimelineEvents', () => {
  const { asFragment } = render(<TimelineEvents
    militaryYears={mockMilitaryYears}
    postServiceYears={mockPostServiceYears}
  />);

  expect(asFragment()).toMatchSnapshot();
});
