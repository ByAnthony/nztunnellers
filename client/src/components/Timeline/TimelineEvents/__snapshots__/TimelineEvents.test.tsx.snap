// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TimelineEvents 1`] = `
<DocumentFragment>
  <div>
    <h2
      aria-label="Year 1915"
      class="year-title"
    >
      1915
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1915
        </div>
        <div
          class="day-month"
        >
          26 January
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Enlisted
          </p>
          <span>
            Main Body
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1915
        </div>
        <div
          class="day-month"
        >
          14 February
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event-with-title"
        >
          <p>
            Trained
          </p>
          <span>
            Military Camp, Somewhere
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1915
        </div>
        <div
          class="day-month"
        >
          4 May
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1916"
      class="year-title"
    >
      1916
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1916
        </div>
        <div
          class="day-month"
        >
          19 November
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1916
        </div>
        <div
          class="day-month"
        >
          12 December
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div>
          <div
            class="company-event"
          >
            <img
              alt=""
              src="/images/roll/image.jpg"
            />
            <p>
              Something happened that day for the Company
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1917"
      class="year-title"
    >
      1917
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1917
        </div>
        <div
          class="day-month"
        >
          9 April
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
        <div>
          <div
            class="company-event"
          >
            <img
              alt=""
              src="/images/roll/image.jpg"
            />
            <p>
              Something happened that day for the Company
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1917
        </div>
        <div
          class="day-month"
        >
          8 August
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1918"
      class="year-title"
    >
      1918
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1918
        </div>
        <div
          class="day-month"
        >
          3 March
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1918
        </div>
        <div
          class="day-month"
        >
          13 October
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1919"
      class="year-title"
    >
      1919
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1919
        </div>
        <div
          class="day-month"
        >
          4 April
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1919
        </div>
        <div
          class="day-month"
        >
          12 July
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render a timeline 1`] = `
<DocumentFragment>
  <div>
    <h2
      aria-label="Year 1915"
      class="year-title"
    >
      1915
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1915
        </div>
        <div
          class="day-month"
        >
          26 January
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Enlisted
          </p>
          <span>
            Main Body
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1915
        </div>
        <div
          class="day-month"
        >
          14 February
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event-with-title"
        >
          <p>
            Trained
          </p>
          <span>
            Military Camp, Somewhere
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1915
        </div>
        <div
          class="day-month"
        >
          4 May
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1916"
      class="year-title"
    >
      1916
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1916
        </div>
        <div
          class="day-month"
        >
          19 November
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1916
        </div>
        <div
          class="day-month"
        >
          12 December
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div>
          <div
            class="company-event"
          >
            <img
              alt=""
              src="/images/roll/image.jpg"
            />
            <p>
              Something happened that day for the Company
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1917"
      class="year-title"
    >
      1917
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1917
        </div>
        <div
          class="day-month"
        >
          9 April
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
        <div>
          <div
            class="company-event"
          >
            <img
              alt=""
              src="/images/roll/image.jpg"
            />
            <p>
              Something happened that day for the Company
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1917
        </div>
        <div
          class="day-month"
        >
          8 August
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1918"
      class="year-title"
    >
      1918
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1918
        </div>
        <div
          class="day-month"
        >
          3 March
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1918
        </div>
        <div
          class="day-month"
        >
          13 October
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h2
      aria-label="Year 1919"
      class="year-title"
    >
      1919
    </h2>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1919
        </div>
        <div
          class="day-month"
        >
          4 April
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="main-event"
        >
          <p>
            Major Event
          </p>
          <span>
            Something happened
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="date"
      >
        <div
          class="year"
        >
          1919
        </div>
        <div
          class="day-month"
        >
          12 July
        </div>
      </div>
      <div
        class="event-wrapper"
      >
        <div
          class="tunneller-event"
        >
          <p>
            Something happened that day
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
